libs/auth/service_principal.go

```diff
@@ -4,13 +4,21 @@ import (
 	"context"
 
 	"github.com/databricks/databricks-sdk-go"
+	"github.com/databricks/databricks-sdk-go/apierr"
 )
 
 // Determines whether a given user id is a service principal.
 // This function uses a heuristic: if no user exists with this id, we assume
 // it's a service principal. Unfortunately, the standard service principal API is too
 // slow for our purposes.
-func IsServicePrincipal(ctx context.Context, ws *databricks.WorkspaceClient, userId string) bool {
+func IsServicePrincipal(ctx context.Context, ws *databricks.WorkspaceClient, userId string) (bool, error) {
 	_, err := ws.Users.GetById(ctx, userId)
-	return err != nil
+	if err != nil {
+		if apiError, ok := err.(*apierr.APIError); ok {
+			if apiError.StatusCode == 404 {
+				return true, nil
+			}
+		}
```

Comment:
Please use `apierr.IsMissing` instead.

Note that type casting the API error is not always correct, as the error may be wrapped.

Also see https://pkg.go.dev/errors#As